app/controllers/root_proxies_controller.rb

```diff
@@ -1,43 +1,43 @@
 # frozen_string_literal: true
 
 class RootProxiesController < ApplicationController
-  def index
-    render jsonapi: root_proxies.all, include: params[:include]
+  deserializable_resource :root_proxy, only: [:update],
+                                       class: DeserializableRootProxy
+
+  def show
+    @result = root_proxy.find!
+    render jsonapi: @result.data
   end
 
   def update
-    data_attributes = JSON.parse(request.body.read)['data']['attributes'] || {}
-    root_proxy_id = params[:id]
-
-    root_proxy_validation =
-      Validation::RootProxyParameters.new(data_attributes,
-                                          root_proxy_id,
-                                          root_proxies.all)
+    proxy_types_list = ProxyTypesRepository.new(config: config).all!
```

Comment:
All of this work is the responsibility of the repository.

the controller method should look like

```ruby
def update
  @result = root_proxy.update! root_proxy_update_params
  render jsonapi: :no_content # 204 No Content or however it's done in rails
end
```

everything else, validation, error handling, pre-requisite reads, are all the responsibility of the repository. The controller is only concerned with HTTP.

Reply:
@sadatay said that it was settle upon earlier that because of the side-effects that RMAPI does that we would return a body.

Reply:
^ this.  if we want to be consistent with the other endpoints anyway.  the reasoning behind using a 204 depends on knowing with certainty that a resource + an update = current state.  we don't have that certainty elsewhere in rmapi so we return resource bodies on update everywhere else.